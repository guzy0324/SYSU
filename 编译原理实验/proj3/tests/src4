/* ex4: Arrays */
program ex4;
class c4
{
	declarations
		int[] x=int[10];
		int[] ia[]={3, 5, 7};
	enddeclarations
	method void main()
	declarations
		int[] y=int [25];
		int z ;
	enddeclarations
	{
	y[3] := 17; 
	z := y[3] + 3;
	ia[2] := 27 ; 
	system.println('z=');
	system.println(z); 
	system.println('ia[0]=');
	system.println(ia[0]);
	system.println('z=');
	system.println(z); 
	system.println('ia[1]=');
	system.println(ia[1]); 
	system.println('ia[2]=');
	system.println(ia[2]);
        z := -1;
	while (z<2)
	{
	  ia[z+1] := 1 + z*2 + z*z;
	  z:=z+1;
	};
        z := 0;
	while (z<=2)
	{
	  system.println(ia[z]);
	  z:=z+1;
	};
	}
}
