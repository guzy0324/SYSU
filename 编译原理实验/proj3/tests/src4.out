********************************Symbol Table************************************

          Name Nest-Level  Tree-Node Predefined        Kind      Type      Value     Offset  Dimension     Argnum

  1     system          0                   yes      class                                                       
  2    println          1                   yes  procedure                                                      1
  3         c4          0                            class                                                       
  4          x          1                            array  type-tree                                1           
  5         ia          1                            array  type-tree                                1           
  6       main          1  proc-tree             procedure                                                      0
  7          y          2                            array  type-tree                                1           
  8          z          2                         variable  type-tree                                            
************* SYNTAX TREE PRINTOUT ***********

  +-[DUMMYnode]
R-[ProgramOp]
  |   +-[STNode,3,"c4"]
  | +-[ClassDefOp]
  | | |               +-[NUMNode,1]
  | | |             +-[AddOp]
  | | |             | | +-[DUMMYnode]
  | | |             | +-[VarOp]
  | | |             |   +-[STNode,8,"z"]
  | | |           +-[AssignOp]
  | | |           | |   +-[DUMMYnode]
  | | |           | | +-[VarOp]
  | | |           | | | +-[STNode,8,"z"]
  | | |           | +-[AssignOp]
  | | |           |   +-[DUMMYnode]
  | | |         +-[StmtOp]
  | | |         | |     +-[DUMMYnode]
  | | |         | |   +-[CommaOp]
  | | |         | |   | |   +-[DUMMYnode]
  | | |         | |   | | +-[SelectOp]
  | | |         | |   | | | | +-[DUMMYnode]
  | | |         | |   | | | +-[IndexOp]
  | | |         | |   | | |   | +-[DUMMYnode]
  | | |         | |   | | |   +-[VarOp]
  | | |         | |   | | |     +-[STNode,8,"z"]
  | | |         | |   | +-[VarOp]
  | | |         | |   |   +-[STNode,5,"ia"]
  | | |         | | +-[RoutineCallOp]
  | | |         | | | |   +-[DUMMYnode]
  | | |         | | | | +-[SelectOp]
  | | |         | | | | | | +-[DUMMYnode]
  | | |         | | | | | +-[FieldOp]
  | | |         | | | | |   +-[STNode,2,"println"]
  | | |         | | | +-[VarOp]
  | | |         | | |   +-[STNode,1,"system"]
  | | |         | +-[StmtOp]
  | | |         |   +-[DUMMYnode]
  | | |       +-[LoopOp]
  | | |       | | +-[NUMNode,2]
  | | |       | +-[LEOp]
  | | |       |   | +-[DUMMYnode]
  | | |       |   +-[VarOp]
  | | |       |     +-[STNode,8,"z"]
  | | |     +-[StmtOp]
  | | |     | |   +-[NUMNode,0]
  | | |     | | +-[AssignOp]
  | | |     | | | |   +-[DUMMYnode]
  | | |     | | | | +-[VarOp]
  | | |     | | | | | +-[STNode,8,"z"]
  | | |     | | | +-[AssignOp]
  | | |     | | |   +-[DUMMYnode]
  | | |     | +-[StmtOp]
  | | |     |   |         +-[NUMNode,1]
  | | |     |   |       +-[AddOp]
  | | |     |   |       | | +-[DUMMYnode]
  | | |     |   |       | +-[VarOp]
  | | |     |   |       |   +-[STNode,8,"z"]
  | | |     |   |     +-[AssignOp]
  | | |     |   |     | |   +-[DUMMYnode]
  | | |     |   |     | | +-[VarOp]
  | | |     |   |     | | | +-[STNode,8,"z"]
  | | |     |   |     | +-[AssignOp]
  | | |     |   |     |   +-[DUMMYnode]
  | | |     |   |   +-[StmtOp]
  | | |     |   |   | |         +-[DUMMYnode]
  | | |     |   |   | |       +-[VarOp]
  | | |     |   |   | |       | +-[STNode,8,"z"]
  | | |     |   |   | |     +-[MultOp]
  | | |     |   |   | |     | | +-[DUMMYnode]
  | | |     |   |   | |     | +-[VarOp]
  | | |     |   |   | |     |   +-[STNode,8,"z"]
  | | |     |   |   | |   +-[AddOp]
  | | |     |   |   | |   | |   +-[NUMNode,2]
  | | |     |   |   | |   | | +-[MultOp]
  | | |     |   |   | |   | | | | +-[DUMMYnode]
  | | |     |   |   | |   | | | +-[VarOp]
  | | |     |   |   | |   | | |   +-[STNode,8,"z"]
  | | |     |   |   | |   | +-[AddOp]
  | | |     |   |   | |   |   +-[NUMNode,1]
  | | |     |   |   | | +-[AssignOp]
  | | |     |   |   | | | |     +-[DUMMYnode]
  | | |     |   |   | | | |   +-[SelectOp]
  | | |     |   |   | | | |   | | +-[DUMMYnode]
  | | |     |   |   | | | |   | +-[IndexOp]
  | | |     |   |   | | | |   |   | +-[NUMNode,1]
  | | |     |   |   | | | |   |   +-[AddOp]
  | | |     |   |   | | | |   |     | +-[DUMMYnode]
  | | |     |   |   | | | |   |     +-[VarOp]
  | | |     |   |   | | | |   |       +-[STNode,8,"z"]
  | | |     |   |   | | | | +-[VarOp]
  | | |     |   |   | | | | | +-[STNode,5,"ia"]
  | | |     |   |   | | | +-[AssignOp]
  | | |     |   |   | | |   +-[DUMMYnode]
  | | |     |   |   | +-[StmtOp]
  | | |     |   |   |   +-[DUMMYnode]
  | | |     |   | +-[LoopOp]
  | | |     |   | | | +-[NUMNode,2]
  | | |     |   | | +-[LTOp]
  | | |     |   | |   | +-[DUMMYnode]
  | | |     |   | |   +-[VarOp]
  | | |     |   | |     +-[STNode,8,"z"]
  | | |     |   +-[StmtOp]
  | | |     |     |     +-[DUMMYnode]
  | | |     |     |   +-[UnaryNegOp]
  | | |     |     |   | +-[NUMNode,1]
  | | |     |     | +-[AssignOp]
  | | |     |     | | |   +-[DUMMYnode]
  | | |     |     | | | +-[VarOp]
  | | |     |     | | | | +-[STNode,8,"z"]
  | | |     |     | | +-[AssignOp]
  | | |     |     | |   +-[DUMMYnode]
  | | |     |     +-[StmtOp]
  | | |     |       |     +-[DUMMYnode]
  | | |     |       |   +-[CommaOp]
  | | |     |       |   | |   +-[DUMMYnode]
  | | |     |       |   | | +-[SelectOp]
  | | |     |       |   | | | | +-[DUMMYnode]
  | | |     |       |   | | | +-[IndexOp]
  | | |     |       |   | | |   +-[NUMNode,2]
  | | |     |       |   | +-[VarOp]
  | | |     |       |   |   +-[STNode,5,"ia"]
  | | |     |       | +-[RoutineCallOp]
  | | |     |       | | |   +-[DUMMYnode]
  | | |     |       | | | +-[SelectOp]
  | | |     |       | | | | | +-[DUMMYnode]
  | | |     |       | | | | +-[FieldOp]
  | | |     |       | | | |   +-[STNode,2,"println"]
  | | |     |       | | +-[VarOp]
  | | |     |       | |   +-[STNode,1,"system"]
  | | |     |       +-[StmtOp]
  | | |     |         |     +-[DUMMYnode]
  | | |     |         |   +-[CommaOp]
  | | |     |         |   | +-[STRINGNode,53,"ia[2]="]
  | | |     |         | +-[RoutineCallOp]
  | | |     |         | | |   +-[DUMMYnode]
  | | |     |         | | | +-[SelectOp]
  | | |     |         | | | | | +-[DUMMYnode]
  | | |     |         | | | | +-[FieldOp]
  | | |     |         | | | |   +-[STNode,2,"println"]
  | | |     |         | | +-[VarOp]
  | | |     |         | |   +-[STNode,1,"system"]
  | | |     |         +-[StmtOp]
  | | |     |           |     +-[DUMMYnode]
  | | |     |           |   +-[CommaOp]
  | | |     |           |   | |   +-[DUMMYnode]
  | | |     |           |   | | +-[SelectOp]
  | | |     |           |   | | | | +-[DUMMYnode]
  | | |     |           |   | | | +-[IndexOp]
  | | |     |           |   | | |   +-[NUMNode,1]
  | | |     |           |   | +-[VarOp]
  | | |     |           |   |   +-[STNode,5,"ia"]
  | | |     |           | +-[RoutineCallOp]
  | | |     |           | | |   +-[DUMMYnode]
  | | |     |           | | | +-[SelectOp]
  | | |     |           | | | | | +-[DUMMYnode]
  | | |     |           | | | | +-[FieldOp]
  | | |     |           | | | |   +-[STNode,2,"println"]
  | | |     |           | | +-[VarOp]
  | | |     |           | |   +-[STNode,1,"system"]
  | | |     |           +-[StmtOp]
  | | |     |             |     +-[DUMMYnode]
  | | |     |             |   +-[CommaOp]
  | | |     |             |   | +-[STRINGNode,46,"ia[1]="]
  | | |     |             | +-[RoutineCallOp]
  | | |     |             | | |   +-[DUMMYnode]
  | | |     |             | | | +-[SelectOp]
  | | |     |             | | | | | +-[DUMMYnode]
  | | |     |             | | | | +-[FieldOp]
  | | |     |             | | | |   +-[STNode,2,"println"]
  | | |     |             | | +-[VarOp]
  | | |     |             | |   +-[STNode,1,"system"]
  | | |     |             +-[StmtOp]
  | | |     |               |     +-[DUMMYnode]
  | | |     |               |   +-[CommaOp]
  | | |     |               |   | | +-[DUMMYnode]
  | | |     |               |   | +-[VarOp]
  | | |     |               |   |   +-[STNode,8,"z"]
  | | |     |               | +-[RoutineCallOp]
  | | |     |               | | |   +-[DUMMYnode]
  | | |     |               | | | +-[SelectOp]
  | | |     |               | | | | | +-[DUMMYnode]
  | | |     |               | | | | +-[FieldOp]
  | | |     |               | | | |   +-[STNode,2,"println"]
  | | |     |               | | +-[VarOp]
  | | |     |               | |   +-[STNode,1,"system"]
  | | |     |               +-[StmtOp]
  | | |     |                 |     +-[DUMMYnode]
  | | |     |                 |   +-[CommaOp]
  | | |     |                 |   | +-[STRINGNode,36,"z="]
  | | |     |                 | +-[RoutineCallOp]
  | | |     |                 | | |   +-[DUMMYnode]
  | | |     |                 | | | +-[SelectOp]
  | | |     |                 | | | | | +-[DUMMYnode]
  | | |     |                 | | | | +-[FieldOp]
  | | |     |                 | | | |   +-[STNode,2,"println"]
  | | |     |                 | | +-[VarOp]
  | | |     |                 | |   +-[STNode,1,"system"]
  | | |     |                 +-[StmtOp]
  | | |     |                   |     +-[DUMMYnode]
  | | |     |                   |   +-[CommaOp]
  | | |     |                   |   | |   +-[DUMMYnode]
  | | |     |                   |   | | +-[SelectOp]
  | | |     |                   |   | | | | +-[DUMMYnode]
  | | |     |                   |   | | | +-[IndexOp]
  | | |     |                   |   | | |   +-[NUMNode,0]
  | | |     |                   |   | +-[VarOp]
  | | |     |                   |   |   +-[STNode,5,"ia"]
  | | |     |                   | +-[RoutineCallOp]
  | | |     |                   | | |   +-[DUMMYnode]
  | | |     |                   | | | +-[SelectOp]
  | | |     |                   | | | | | +-[DUMMYnode]
  | | |     |                   | | | | +-[FieldOp]
  | | |     |                   | | | |   +-[STNode,2,"println"]
  | | |     |                   | | +-[VarOp]
  | | |     |                   | |   +-[STNode,1,"system"]
  | | |     |                   +-[StmtOp]
  | | |     |                     |     +-[DUMMYnode]
  | | |     |                     |   +-[CommaOp]
  | | |     |                     |   | +-[STRINGNode,39,"ia[0]="]
  | | |     |                     | +-[RoutineCallOp]
  | | |     |                     | | |   +-[DUMMYnode]
  | | |     |                     | | | +-[SelectOp]
  | | |     |                     | | | | | +-[DUMMYnode]
  | | |     |                     | | | | +-[FieldOp]
  | | |     |                     | | | |   +-[STNode,2,"println"]
  | | |     |                     | | +-[VarOp]
  | | |     |                     | |   +-[STNode,1,"system"]
  | | |     |                     +-[StmtOp]
  | | |     |                       |     +-[DUMMYnode]
  | | |     |                       |   +-[CommaOp]
  | | |     |                       |   | | +-[DUMMYnode]
  | | |     |                       |   | +-[VarOp]
  | | |     |                       |   |   +-[STNode,8,"z"]
  | | |     |                       | +-[RoutineCallOp]
  | | |     |                       | | |   +-[DUMMYnode]
  | | |     |                       | | | +-[SelectOp]
  | | |     |                       | | | | | +-[DUMMYnode]
  | | |     |                       | | | | +-[FieldOp]
  | | |     |                       | | | |   +-[STNode,2,"println"]
  | | |     |                       | | +-[VarOp]
  | | |     |                       | |   +-[STNode,1,"system"]
  | | |     |                       +-[StmtOp]
  | | |     |                         |     +-[DUMMYnode]
  | | |     |                         |   +-[CommaOp]
  | | |     |                         |   | +-[STRINGNode,36,"z="]
  | | |     |                         | +-[RoutineCallOp]
  | | |     |                         | | |   +-[DUMMYnode]
  | | |     |                         | | | +-[SelectOp]
  | | |     |                         | | | | | +-[DUMMYnode]
  | | |     |                         | | | | +-[FieldOp]
  | | |     |                         | | | |   +-[STNode,2,"println"]
  | | |     |                         | | +-[VarOp]
  | | |     |                         | |   +-[STNode,1,"system"]
  | | |     |                         +-[StmtOp]
  | | |     |                           |   +-[NUMNode,27]
  | | |     |                           | +-[AssignOp]
  | | |     |                           | | |     +-[DUMMYnode]
  | | |     |                           | | |   +-[SelectOp]
  | | |     |                           | | |   | | +-[DUMMYnode]
  | | |     |                           | | |   | +-[IndexOp]
  | | |     |                           | | |   |   +-[NUMNode,2]
  | | |     |                           | | | +-[VarOp]
  | | |     |                           | | | | +-[STNode,5,"ia"]
  | | |     |                           | | +-[AssignOp]
  | | |     |                           | |   +-[DUMMYnode]
  | | |     |                           +-[StmtOp]
  | | |     |                             |     +-[NUMNode,3]
  | | |     |                             |   +-[AddOp]
  | | |     |                             |   | |   +-[DUMMYnode]
  | | |     |                             |   | | +-[SelectOp]
  | | |     |                             |   | | | | +-[DUMMYnode]
  | | |     |                             |   | | | +-[IndexOp]
  | | |     |                             |   | | |   +-[NUMNode,3]
  | | |     |                             |   | +-[VarOp]
  | | |     |                             |   |   +-[STNode,7,"y"]
  | | |     |                             | +-[AssignOp]
  | | |     |                             | | |   +-[DUMMYnode]
  | | |     |                             | | | +-[VarOp]
  | | |     |                             | | | | +-[STNode,8,"z"]
  | | |     |                             | | +-[AssignOp]
  | | |     |                             | |   +-[DUMMYnode]
  | | |     |                             +-[StmtOp]
  | | |     |                               |   +-[NUMNode,17]
  | | |     |                               | +-[AssignOp]
  | | |     |                               | | |     +-[DUMMYnode]
  | | |     |                               | | |   +-[SelectOp]
  | | |     |                               | | |   | | +-[DUMMYnode]
  | | |     |                               | | |   | +-[IndexOp]
  | | |     |                               | | |   |   +-[NUMNode,3]
  | | |     |                               | | | +-[VarOp]
  | | |     |                               | | | | +-[STNode,7,"y"]
  | | |     |                               | | +-[AssignOp]
  | | |     |                               | |   +-[DUMMYnode]
  | | |     |                               +-[StmtOp]
  | | |     |                                 +-[DUMMYnode]
  | | |   +-[BodyOp]
  | | |   | |       +-[DUMMYnode]
  | | |   | |     +-[CommaOp]
  | | |   | |     | | +-[DUMMYnode]
  | | |   | |     | +-[TypeIdOp]
  | | |   | |     |   +-[INTEGERTNode]
  | | |   | |   +-[CommaOp]
  | | |   | |   | +-[STNode,8,"z"]
  | | |   | | +-[DeclOp]
  | | |   | | | +-[DUMMYnode]
  | | |   | +-[BodyOp]
  | | |   |   |         +-[INTEGERTNode]
  | | |   |   |       +-[ArrayTypeOp]
  | | |   |   |       | | +-[NUMNode,25]
  | | |   |   |       | +-[BoundOp]
  | | |   |   |       |   +-[DUMMYnode]
  | | |   |   |     +-[CommaOp]
  | | |   |   |     | |   +-[DUMMYnode]
  | | |   |   |     | | +-[IndexOp]
  | | |   |   |     | | | +-[DUMMYnode]
  | | |   |   |     | +-[TypeIdOp]
  | | |   |   |     |   +-[INTEGERTNode]
  | | |   |   |   +-[CommaOp]
  | | |   |   |   | +-[STNode,7,"y"]
  | | |   |   | +-[DeclOp]
  | | |   |   | | +-[DUMMYnode]
  | | |   |   +-[BodyOp]
  | | |   |     +-[DUMMYnode]
  | | | +-[MethodOp]
  | | | | |   +-[DUMMYnode]
  | | | | | +-[SpecOp]
  | | | | | | +-[DUMMYnode]
  | | | | +-[HeadOp]
  | | | |   +-[STNode,6,"main"]
  | | +-[BodyOp]
  | |   |             +-[DUMMYnode]
  | |   |           +-[IndexOp]
  | |   |           | +-[DUMMYnode]
  | |   |         +-[TypeIdOp]
  | |   |         | +-[INTEGERTNode]
  | |   |       +-[ArrayTypeOp]
  | |   |       | | +-[NUMNode,7]
  | |   |       | +-[CommaOp]
  | |   |       |   | +-[NUMNode,5]
  | |   |       |   +-[CommaOp]
  | |   |       |     | +-[NUMNode,3]
  | |   |       |     +-[CommaOp]
  | |   |       |       +-[DUMMYnode]
  | |   |     +-[CommaOp]
  | |   |     | |   +-[DUMMYnode]
  | |   |     | | +-[IndexOp]
  | |   |     | | | +-[DUMMYnode]
  | |   |     | +-[TypeIdOp]
  | |   |     |   +-[INTEGERTNode]
  | |   |   +-[CommaOp]
  | |   |   | +-[STNode,5,"ia"]
  | |   | +-[DeclOp]
  | |   | | +-[DUMMYnode]
  | |   +-[BodyOp]
  | |     |         +-[INTEGERTNode]
  | |     |       +-[ArrayTypeOp]
  | |     |       | | +-[NUMNode,10]
  | |     |       | +-[BoundOp]
  | |     |       |   +-[DUMMYnode]
  | |     |     +-[CommaOp]
  | |     |     | |   +-[DUMMYnode]
  | |     |     | | +-[IndexOp]
  | |     |     | | | +-[DUMMYnode]
  | |     |     | +-[TypeIdOp]
  | |     |     |   +-[INTEGERTNode]
  | |     |   +-[CommaOp]
  | |     |   | +-[STNode,4,"x"]
  | |     | +-[DeclOp]
  | |     | | +-[DUMMYnode]
  | |     +-[BodyOp]
  | |       +-[DUMMYnode]
  +-[ClassOp]
    +-[DUMMYnode]
